Merge LED branch

Keobkeig opened the pull request.

Keobkeig committed:
add led subsystem

Keobkeig committed:
Need to migrate LEDOdometry into an LED commands

Keobkeig committed:
finish LEDAlign.java needs testing

Keobkeig committed:
add LEDREADME.md

Keobkeig committed:
prep LEDAutonChooser to be written by newbies

Keobkeig committed:
FINISHHHED HEHEHEHAW add LEDAlign

Keobkeig committed:
Merge branch 'leds' of https://github.com/StuyPulse/BigWang into leds

Keobkeig committed:
made disabled LED commands toggleable smart bools, cleaned up LEDAutonChooser, moved new instruction to LEDColor

Keobkeig committed:
-Wrote LEDSetContinous to purple, binded to driver to call for HP's attention
-Make note that LEDSetContinous should be used with whileTrue binding as opposed to setLED being only onTrue/False bindings

Keobkeig committed:
Update LEDSetContinous.java

Keobkeig committed:
Final Commit for LEDs
-remove redundancy between LEDSet and LEDSetContinous

Keobkeig committed:
Merge branch 'main' into leds

Keobkeig committed:
Fixed all merge conflicts for PR

anivanchen committed:
Move led documentation

BenG49 commented on src/main/java/com/stuypulse/robot/subsystems/leds/instructions/LEDSection.java:
Add int array parameter of divider indexes, where each section ends at its divider index

BenG49 replied:
Can have a constructor overload that takes in a color list and makes equally spaced sections array

BenG49 commented on src/main/java/com/stuypulse/robot/subsystems/leds/instructions/LEDPulseColor.java:
Make pulse time a parameter, make an overload of constructor that has default pulse time

BenG49 commented on src/main/java/com/stuypulse/robot/subsystems/leds/instructions/LEDAutonChooser.java:
Add SLColor[] field to the autons enum, initialize in constructor

Rename AUTONS to AutonLEDColors or something

Keobkeig committed:
MOSTLY FIXED FROM CODE REVIEW
-LEDColor now LEDInstructions
-ledpulsecolor now can have a manually set blink time
-AutonChooser logic mostly refactored into enum (+ edge case support for DoNothingAuton)
-LEDController rid javadoc and change forceSetLED to runLEDInstruction
- rid unused manual update time in Settings
- moved LEDAlign into instruction folder
- disabled commands are now exclusively ran via SmartBoolean
- made setDisabledLED for disabled LED commands

BenG49 committed:
Undo json changes

BenG49 committed:
Refactoring

BenG49 committed:
Add LEDSingleColor constructors

Keobkeig committed:
Revert refactoring of LEDAutonChooser

BenG49 committed:
Remove enum map from auton chooser

Material-Energy committed:
fixing auton chooser

added match name method instead of valueOf,
uses arrays.fill instead of manually checking afterwards

alex-wanger committed:
add LEDSectionNew

Material-Energy committed:
reworked LEDSection

to allow for custom lengthed sections of color. additionally, indexes can be omitted and they will be evenly distributed.

Material-Energy committed:
Merge branch 'leds' of https://github.com/StuyPulse/Izzi into leds

Material-Energy committed:
Removed LEDSectionNew

kind of redundant

Keobkeig committed:
Fix code rev

Co-authored-by: naveed1117 <naveed1117@users.noreply.github.com>
Co-authored-by: Zixi52 <Zixi52@users.noreply.github.com>
Co-authored-by: Owen-Zeng <Owen-Zeng@users.noreply.github.com>
Co-authored-by: Rahuldeb75 <Rahuldeb75@users.noreply.github.com>
Co-authored-by: BrianZ60 <BrianZ60@users.noreply.github.com>
Co-authored-by: Edgar694 <Cancelable@users.noreply.github.com>

Keobkeig committed:
rev unused imports

Keobkeig committed:
add binding to operator select button

BenG49 committed:
Review changes

BenG49 committed:
Make led set back to rainbow after button

BenG49 committed:
Merge branch 'main' into leds
